RadixWallet/Features/AccountRecoveryScan/Children/AccountRecoveryScanInProgress/AccountRecoveryScanInProgress.swift

```diff
@@ -112,49 +136,75 @@ public struct AccountRecoveryScanInProgress: Sendable, FeatureReducer {
+			case .privateHD:
+				return derivePublicKeys(state: &state)
+			case .factorSourceWithID:
 				state.status = .loadingFactorSource
-				return .run { [id = state.factorSourceID] send in
-					let result = await TaskResult<FactorSource?> {
-						try await factorSourcesClient.getFactorSource(id: id.embed())
+				let id = state.factorSourceIDFromHash
+				state.mode = .factorSourceWithID(id: id, .loading)
+				return .run { [networkID = state.networkID] send in
+					let result = await TaskResult<IndicesUsedByFactorSource> {
+						try await factorSourcesClient.indicesOfEntitiesControlledByFactorSource(
+							.init(
+								entityKind: .account,
+								factorSourceID: id.embed(),
+								networkID: networkID
+							)
+						)
 					}
-					await send(.internal(.loadFactorSourceResult(result)))
+					await send(.internal(.loadIndicesUsedByFactorSourceResult(result)))
 				}
 			}
 
 		case .scanMore:
-			guard let factorSource = state.factorSource.wrappedValue else { fatalError("discrepancy") }
-			state.offset += accRecScanBatchSize
-			return derivePublicKeys(using: factorSource, state: &state)
+			loggerGlobal.debug("Scan more requested.")
+			return derivePublicKeys(state: &state)
 
 		case .continueTapped:
-			return .send(.delegate(.foundAccounts(active: state.active, inactive: state.inactive)))
+			if let maxActive = state.active.max() {
+				let inactiveInBetweenActive = state.inactive.filter {
+					$0.derivationIndex < maxActive.derivationIndex
+				}
+				return .send(.delegate(.foundAccounts(active: state.active, inactive: inactiveInBetweenActive)))
+			} else {
+				return .send(.delegate(.foundAccounts(active: [], inactive: [])))
+			}
 		}
 	}
 
-	public func reduce(into state: inout State, childAction: ChildAction) -> Effect<Action> {
-		switch childAction {
-		/// Attention! If you change this to the `destination` pattern we use elsewhere this feature breaks due to "TCA Send"-bug
-		case let .derivePublicKeys(.presented(.delegate(delegateAction))):
-			loggerGlobal.notice("Finish deriving public keys")
+	public func reduce(into state: inout State, presentedAction: Destination.Action) -> Effect<Action> {
+		switch presentedAction {
+		case let .derivePublicKeys(.delegate(delegateAction)):
+			loggerGlobal.notice("Finish deriving public keys => `state.destination = nil`")
 			switch delegateAction {
 			case let .derivedPublicKeys(publicHDKeys, factorSourceID, networkID):
-				assert(factorSourceID == state.factorSourceID.embed())
+				let id = state.factorSourceIDFromHash
+				assert(factorSourceID == id.embed())
 				assert(networkID == state.networkID)
-				return .run { [id = state.factorSourceID] send in
-					let accounts = await publicHDKeys.enumerated().asyncMap { offset, publicHDKey in
+				return .run { send in
+					let accounts = await publicHDKeys.enumerated().asyncMap {
+						offset,
+							publicHDKey in
 						let appearanceID = await accountsClient.nextAppearanceID(networkID, offset)
```

Comment:
this is a bug right? we need another "global" offset for appearance ID (since 12 (number of gradients) does not divide 50 (accountsBatchSize))! so hmm `accountsClient.nextAppearanceID(networkID, offset + state.activeAccounts.count + state.inactiveAccounts.count)` right?

Reply:
Is offset only local to the batch? Then probably yes. I am not sure what behaviour we actually want, should it be based on active accounts or all accounts? If all accounts, then maybe just take the `(batchNumber * batchSize + + offset) % 12`, if we know which batch we are in.

Reply:
fixed in https://github.com/radixdlt/babylon-wallet-ios/pull/935